### ISSUE TITLE
Assets won't reflect the choosen container in the saved information

### ISSUE THREAD

**Issue by jonassiewertsen** [statamic/cms#2119]
There are some issues to find Assets via the Asset facade. After some dinging and tinkering I came to the conclusion that the problem is the saved path of the asset. **The saved path does not reflect the container.**

The docs do state, that a container does not need to be defined if only one container exists. Anyway. To keep it simple, I will explicit define the container. 

<img width="640" alt="Screenshot 2020-07-21 at 23 04 39" src="https://user-images.githubusercontent.com/38906163/88108076-884e5280-cba8-11ea-88f2-a7f8a646e98c.png">

Statamic itself does handle this case somehow. It's getting tricky if working in addon context or extending a plain Statamic page while working with the Asset facade. 

A blueprint to save images
```
handle: assets
        field:
          mode: grid
          container: assets
          restrict: false
          allow_uploads: true
          type: assets
          localizable: false
          listable: hidden
          display: Assets
```

As we can see, the container has been set. 

If saving images, we'll get the following data from the .md file as created by Statamic.

```
assets:
  - products/echeveria-agavoides.jpg
```

My directory:

<img width="300" alt="directory" src="https://user-images.githubusercontent.com/38906163/88108719-9c468400-cba9-11ea-844b-a3e2e763da32.png">

If receiving the assets directly from a collection (This could be in an addon context as well), we will get `products/echeveria-agavoides.jpg`. The path does not reflect the container via path or container definition. 

Why does it matter?

```
Asset::find('products/echeveria-agavoides.jpg'); // Does return null. The container context is missing. 

// better
Asset::find('assets/products/echeveria-agavoides.jpg'); // returning the asset
Asset::find('assets::products/echeveria-agavoides.jpg'); // returning the asset
```

I would expect the assets to be saved like this instead:
````
// Instead of 
assets:
  - products/echeveria-agavoides.jpg

// More clear
assets:
  - assets/products/echeveria-agavoides.jpg
// or
  - assets::products/echeveria-agavoides.jpg
```

Especially if more containers have been set: Wouldn't this be the fastest and cleanest way to determine, in which container the assets are? 

Those are my thoughts and finds in this case.

## Environment

**Statamic version:** 3.0.0.-beta-38

**PHP version:** 7.4

**Install method (choose one):**
- Fresh install from `statamic/statamic`

**Comment by jonassiewertsen** [comment]
The issues above have been closed because I think this issue is much clearer than the closed ones. 

I hope it does help :-)

**Comment by jonassiewertsen** [comment]
It's expected behavior as I learned from @jasonvarga today. 

The container information needs to be pulled from the blueprint. Instead of working isolated with the relative path, the data should e augmented and an asset object will be returned automatically.

**Comment by fitzage** [comment]
Is this why you can no longer choose from multiple asset containers in an asset field?

This seems like a big step backwards. Also, as much as I hate to second guess people smarter than me, coupling this to the blueprint seems like the wrong move.

**Comment by jasonvarga** [comment]
If you want a particular asset field to be able to choose assets from multiple containers, why do you have separate containers?

**Comment by fitzage** [comment]
1) If I upload an image for an event (for example), and then want to use that image somewhere else, it makes little sense to have to add the image a second time.

2) Sometimes I want to default to a specific location, but allow selecting from another location if desired.

Overarching reason: avoid having to include images in the system multiple times in certain scenarios.

**Comment by jasonvarga** [comment]
Multiple fields can point to the same container. If you upload in one field, you can select it from another field.

**Comment by fitzage** [comment]
Yes, but that means that I can’t have like a featured area where I can pull an image from multiple different asset containers. One time it might be that I want to pull an image from an event. Another time it might be that I want to pull it from a Blog article.

Using a field that allows me to select entries satisfies some of these situations, but not all of them.

**Comment by jasonvarga** [comment]
You should be able to use one container and multiple folders.
Each field can point to a different folder, and restrict navigation so users can't go wondering.
Then your featured field can just have no folder restricted and pick anywhere from the container.

Multiple containers are typically useful for storing in different filesystems, storing different types of files (like pdfs vs. images)

Anyway if you want to open a feature request for asset fields to allow multiple containers, that's cool too.

**Comment by fitzage** [comment]
If I’d known this was going to be an issue back when I set up my v2 site, I would have just gone with separate folders. It would be a lot of work to switch to that now, though.

That also prevents having per-container fieldsets for images, although I decided against using image fields for most things, so that’s probably not a big deal. That’s part of the reason I went with separate containers in the first place.
